bookshelf/src/DatastoreModel.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Samples\Bookshelf;
+
+use Google_Service_Datastore;
+
+/**
+ * Class DatastoreModel implements the DataModel with a Google Data Store.
+ *
+ * Incomplete and untested.
+ */
+class DatastoreModel implements DataModelInterface
+{
+    public function __construct($datastoreDatasetId = null)
+    {
+        if (!$datastoreDatasetId) {
+            $datastoreDatasetId = getenv('GOOGLE_PROJECT_ID');
+        }
+        $this->datasetId = $datastoreDatasetId;
+        $client = new \Google_Client();
+        $client->setScopes([
```

Comment:
I don't think we still need these scopes if it's running on Managed VMs. Do we still need them for other cases?